xcode/Subconscious/Shared/Components/Detail/MemoViewerDetailView.swift

```diff
@@ -464,20 +472,45 @@ struct MemoViewerDetailModel: ModelProtocol {
         environment: Environment,
         description: MemoViewerDetailDescription
     ) -> Update<Self> {
+        guard state.address != description.address else {
+            logger.warning("Attempted to appear with same address, doing nothing")
+            return Update(state: state)
+        }
+        
         var model = state
-        model.loadingState = .loading
         model.address = description.address
         
+        return update(
+            state: model,
+            // Set meta sheet address as well
+            actions: [
+                .setMetaSheetAddress(description.address),
+                .refreshAll
+            ],
+            environment: environment
+        )
+    }
+    
+    static func refreshAll(
+        state: Self,
+        environment: Environment
+    ) -> Update<Self> {
+        guard let address = state.address else {
+            logger.warning("Attempted to refresh with nil address")
```

Comment:
`logger.log` seems more appropriate here, since this is expected behavior that is being handled gracefully.